docs/architecture.md

```diff
@@ -0,0 +1,23 @@
+# Architecture
+
+## The chosen architecture
+libcal-proxy is a [vert.x](https://vertx.io/) application.
+It consists of:
+* A verticle that 
+** Loads application configuration
+** Retrieves initial OAuth tokens
+** Deploys the two request handlers
+* A status handler, called to determine if the application is running
+* A LibCal Proxy handler, which
+** Receives LibCal API requests
+** Retrieves an OAuth token from application storage
+** Passes the API request along to LibCal with the token
+** Returns the LibCal response to the original client
+
+## Functional Overview
+The proxy exists to insulate cients from the process of retrieving an OAuth token from the LibCal OAuth provider. Client calls are identical to direct calls to a LibCal API, except for two points: 
+1) "proxy" is prefixed to the UCLA LibCal URL
```

Comment:
This first point is not true; the request URLs are identical

Reply:
I'm not sure I understand, Mark. Aren't these two URLs different?

https://proxy.calendar.library.ucla.edu/1.1/event_search?search=*
https://calendar.library.ucla.edu/1.1/event_search?search=*

The paths and query strings are identical, but the URLs are different.

Reply:
Added example URLs to clarify

Reply:
yes, obviously those two URLs are different. The DNS resolution of the service is not really part of our implementation so I didn't think that's what was meant here, I thought it was referring to everything after the hostname

Reply:
I should have been more precise with my initial comment

Reply:
From the client's perspective, it's functionally relevant so I think the distinction is good.

Reply:
Agree that it's good to document, but it's already in the README. My understanding of the ARCHITECTURE document is that it should be more about the implementation, rather than describing usage. I'm reading: https://matklad.github.io/2021/02/06/ARCHITECTURE.md.html

Reply:
I think endpoints/API are part of architecture.